tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -3628,9 +2447,12 @@ void FragmentMetadata::load_v1_v2(
   load_non_empty_domain(deserializer);
   load_mbrs(deserializer);
   load_bounding_coords(deserializer);
-  load_tile_offsets(deserializer);
-  load_tile_var_offsets(deserializer);
-  load_tile_var_sizes(deserializer);
+  static_cast<V1V2PreloadedFragmentMetadata*>(offsets_metadata_)
```

Comment:
I'd rather do a dynamic cast here and make sure the pointer is not null. That way we ensure that we created the correct object.